### PR TITLE
chore: bump to argocd 2.0-rc & kubernetes 0.20.4

### PR DESCRIPTION
**Pull request by sbose78**
* Getting as close to ArgoCD master as possible.
* The repo server client config now requires tlsconfig. Configuring the same is outside the scope of this PR.

Signed-off-by: Shoubhik Bose <shbose@redhat.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -77,6 +79,7 @@ func main() {
 			"Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&namespace, "namespace", "", "Argo CD repo namespace (default: argocd)")
 	flag.StringVar(&argocdRepoServer, "argocd-repo-server", "argocd-repo-server:8081", "Argo CD repo server address")
+	flag.IntVar(&argocdRepoServerTimeout, "argocd-repo-server-timeout", 60, "Argo CD repo server connection timeout")
```
Comment (wtam2018) [inline main.go:82]: Unit is in seconds?   It's probably helpful to state the unit in the description.

**Comment by jgwest** [comment]
FYI @sbose78 I'm intentionally keeping this ApplicationSet v0.1.0 release on a fixed version of v1.8.x, so as to continue to support* both v1.8 and v2.0 versions with ApplicationSet controller v0.1.0, and so as not to regress existing functionality.

We can definitely merge this after v0.1.0 is out.

* = support in the sense of automated and manual tests against these versions, to varying degrees.

**Comment by sbose78** [comment]
Sounds perfect!

**Comment by jgwest** [comment]
(And ApplicationSet controller no longer communicates with or uses the repo server, hence why we have not had to add support for the new repo server client TLS configuration in Argo CD, I reviewed this back when @jannfis was proposing the Argo CD PR)

**Comment by sbose78** [comment]
Gotcha, we can skip exposing config for that then. I'll update the PR.

**Comment by jgwest** [comment]
Issue opened to track the work in next milestone: https://github.com/argoproj-labs/applicationset/issues/174

**Comment by jgwest** [comment]
This was handled in https://github.com/argoproj-labs/applicationset/pull/281

**Pull request closed without merging**